code/area.dm

```diff
@@ -486,6 +489,44 @@ TYPEINFO(/area)
 	proc/store_biome(turf/T, datum/biome/B)
 		return
 
+	proc/add_permafrost()
+		src.permafrosted = TRUE
+
+		for (var/turf/simulated/floor/T in src)
+			T.type_before_permafrost = T.type
+			T.icon_before_permafrost = T.icon
+			T.icon_state_before_permafrost = T.icon_state
+			if (T.intact && !istype(T, /turf/simulated/floor/glassblock))
+				T.ReplaceWith(/turf/simulated/floor/snow/snowball)
+				T.icon = 'icons/turf/snow.dmi'
+				T.icon_state = "snow[pick(1, 2, 3)]"
+				T.set_dir(pick(cardinal))
+			else
+				T.icon = 'icons/turf/floors.dmi'
+				T.icon_state = "snow[pick(null, 1, 2, 3, 4)]"
+				T.set_dir(pick(cardinal))
+
+			new /obj/effects/precipitation/snow/grey/tile/light(T)
+
+			T.temperature = T0C
+			T.air.temperature = T0C
+
+			LAGCHECK(LAG_LOW)
+
+	proc/remove_permafrost()
```

Comment:
Removing permafrost doesn't appear to work, random tiles get replaced with space, atmos throws 600 runtimes and the snow remains.